lib/akamai_rspec/matchers/honour_origin_headers.rb

```diff
@@ -30,7 +30,16 @@ def clean_cc_directives(origin_response, akamai_response)
   origin_cc_directives = origin_response.headers[:cache_control].split(/[, ]+/).to_set
   akamai_cc_directives = akamai_response.headers[:cache_control].split(/[, ]+/).to_set
 
-  origin_cc_directives.delete 'must-revalidate' # as Akamai does no pass it on
+  unwanted_values = %w(
+    must-revalidate
+    stale-if-error
+    stale-while-revalidate
+  )
+
+  unwanted_values.each do |unwanted_value|
+    origin_cc_directives.delete_if { |key, _| key.include?(unwanted_value) }
+  end
```

Comment:
This loop can be replaced with:

```ruby
origin_cc_directives.subtract(unwanted_values)
```

Reply:
We're unable to use `subtract` here due to the TTLs on the directives not making exact string matches (and we shouldn't hardcode those values as they will shift). 

```
require 'set'

>> headers = ["private", "max-age=300", "stale-while-invalidate=10"].to_set
=> #<Set: {"private", "max-age=300", "stale-while-invalidate=10"}>

>> headers.subtract(['stale-while-invalidate'])
=> #<Set: {"private", "max-age=300", "stale-while-invalidate=10"}>
```

If it was just the previous values of "must-revalidate" or complete values it would all be good however here we need to fuzzy match and remove the whole value.

Reply:
Missed that logic 😊